light-system-programs/programs/verifier_program_two/src/lib.rs

```diff
@@ -45,12 +58,52 @@ pub mod verifier_program_two {
             b: proof_b,
             c: proof_c,
         };
-        process_shielded_transfer(ctx, &proof, &connecting_hash)?;
-        Ok(())
+
+        let verifier_state = VerifierState10Ins::<2, TransactionConfig>::deserialize(
+            &mut &*ctx.accounts.verifier_state.to_account_info().data.take(),
+        )?;
+
+        let public_amount = Amounts {
+            sol: verifier_state.public_amount_sol,
+            spl: verifier_state.public_amount_spl,
+        };
+
+        if *ctx.accounts.verifier_state.owner == ctx.accounts.system_program.key() {
+            return err!(crate::ErrorCode::InvalidVerifier);
+        };
+
+        let mut owner_hash = hash(&ctx.accounts.verifier_state.owner.to_bytes()).to_bytes();
+        owner_hash[0] = 0;
+        let checked_inputs = [owner_hash, connecting_hash];
+        let leaves = [
+            [verifier_state.leaves[0], verifier_state.leaves[1]],
+            [verifier_state.leaves[2], verifier_state.leaves[3]],
+        ];
+
+        let nullifiers: [[u8; 32]; 4] = verifier_state.nullifiers.to_vec().try_into().unwrap();
+        let pool_type = [0u8; 32];
+        let input = TransactionInput {
+            ctx: &ctx,
+            message: None,
+            proof: &proof,
+            public_amount: &public_amount,
+            checked_public_inputs: &checked_inputs,
+            nullifiers: &nullifiers,
+            leaves: &leaves,
+            encrypted_utxos: &verifier_state.encrypted_utxos.to_vec(),
+            relayer_fee: verifier_state.relayer_fee,
+            merkle_root_index: verifier_state.merkle_root_index as usize,
+            pool_type: &pool_type,
+            verifyingkey: &VERIFYINGKEY,
+        };
+        let mut tx =
+            Transaction::<2, 2, 4, 15, TransactionConfig, LightInstruction<'info>>::new(input);
```

Comment:
Transaction has the following const generics in this order:

``` 
    const NR_CHECKED_INPUTS: usize,
    const NR_LEAVES: usize,
    const NR_NULLIFIERS: usize,
    const NR_PUBLIC_INPUTS: usize,
```

verifier two has 4 in utxos and 4 out utxos, shouldn't it be as follows?
```
Transaction::<2, 4, 4, 15, TransactionConfig, LightInstruction<'info>>::new(input);
```

Reply:
I am thinking how we could avoid such inconsistencies in the future, in particular for PSP devs.
For psp devs we would ideally just provide a verifier_two config struct so that no additional manual config is necessary.

Actually this would be an argument for keeping the TransactionConfig struct and eliminating the const generics.
Then we could have TransactionConfigVerifierTwo which we define in verifier-sdk so that psp devs can import it.

This way we would also get rid of the numbers <2, 2, 4, 15, ...> which is somewhat opaque and I had to look up what they represent.

Reply:
I think the better way to try could be providing the nullifiers and leaves directly as arguments to `new()`, so the compiler is able to figure out the value of const generic and it doesn't have to be provided explicitly.

See this example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=765c5228694d6c28a16d1d48f6990427

Reply:
I'm happy to do that in an another PR

Reply:
cool I didn't know this worked. This is better than the other option